### PR TITLE
Make changes necessary for updating an invoice from client

### PR DESCRIPTION
**Pull request by aaron-suarez**
Allow editing of an invoice from the client.

### CODE REVIEW

File: lib/recurly/invoice.rb [hunk 1]
```diff
@@ -201,6 +202,22 @@ def xml_keys
...
+    # Raises an error if attempting to create an invoice using this method.
+    #
+    # @return [true, false]
+    # @raise [RuntimeError] Raises error if you attempt to create an invoice.
+    # @example
+    #   invoice = Recurly::Invoice.find('1000')
+    #   invoice.po_number = '1234'
+    #   invoice.save # => true
+    def save
+      if !persisted?
```
Comment (bhelx) [inline lib/recurly/invoice.rb]: This is a good use case for `unless`. `unless` is like `if` but it negates the conditional:

```ruby
unless persisted?
  # do something
end
```

File: lib/recurly/invoice.rb [hunk 2]
```diff
@@ -201,6 +202,22 @@ def xml_keys
...
+        raise "New invoices cannot be created using Invoice#save"
```
Comment (bhelx) [inline lib/recurly/invoice.rb]: We might want to choose an error class here, something like:

```ruby
raise BadRequest, "New invoices cannot be created using Invoice#save"
```

File: lib/recurly/invoice.rb (hunk 1 above)
Comment (aaron-suarez) [reply lib/recurly/invoice.rb]: Good catch

File: lib/recurly/invoice.rb [hunk 3]
```diff
@@ -201,6 +202,22 @@ def xml_keys
...
+    #
+    # @return [true, false]
+    # @raise [RuntimeError] Raises error if you attempt to create an invoice.
+    # @example
+    #   invoice = Recurly::Invoice.find('1000')
+    #   invoice.po_number = '1234'
+    #   invoice.save # => true
+    def save
+      unless persisted?
+        raise "Invoices can only be updated with Invoice#save. New invoices cannot be created using this method."
```
Comment (aaron-suarez) [inline lib/recurly/invoice.rb]: Changed error message to clarify that invoices can be updated but not created with this method. Could not use any existing error classes, and it didn't make sense to create a new one at this time.

Comment (bhelx) [reply lib/recurly/invoice.rb]: 👍

**Pull request merged**